DataStorageEngine/DSECommonSenseProxy.h

```diff
@@ -0,0 +1,171 @@
+ */
+- (NSDictionary *) createSensorWithName: (NSString *) name andDisplayName: (NSString *) displayName andDeviceType: (NSString *) deviceType andDataType: (NSString *) dataType andDataStructure: (NSString *) dataStructure andSessionID: (NSString *) sessionID andError: (NSError **) error;
+
+
+/**
+ Get all sensors for the currently logged in user.
+ 
+ This will fetch all the sensors for the current user and pass them on to the success callback as an NSArray. Each element in the array will contain an NSDictionary with data from one sensor.
+ 
+ @param sessionID		The sessionID of the current user. Cannot be empty.
+ @param error			Reference to an NSError object that will contain error information if an error occurs. Cannot be nil.
+ @result				Array of sensors. Each object will be an NSDictionary with the resulting sensor information. Will be nil if an error occurs.
+ */
+- (NSArray *) getSensorsWithSessionID: (NSString *) sessionID andError: (NSError **) error;
+
+
+/**
+ Get all devices for the currently logged in user.
+ 
+ This will fetch all the devices for the current user and pass them on to the success callback as an NSArray. Each element in the array will contain an NSDictionary with data from one device.
+ 
+ @param sessionID		The sessionID of the current user. Cannot be empty.
+ @param error			Reference to an NSError object that will contain error information if an error occurs. Cannot be nil.
+ @result				Array of devices. Each object will be an NSDictionary with the resulting device information. Will be nil if an error occurs.
+ */
+- (NSArray *) getDevicesWithSessionID: (NSString *) sessionID andError: (NSError **) error;
+
+
+
+/**
+ Add sensor to a device.
+ 
+ This will create a device if it doesnt exist yet. This is the only way to create a new device. A device is uniquely identified by a device ID or by a name and UUID. This method will use the name and UUID. There is a twin method available that takes a device ID. If the device ID is available this is the preferred method to use.
+ 
+ @warning iOS does not provide consistent UUIDs anymore. Whenever an app is deinstalled and later reinstalled on the same device, the UUID retrieved from iOS might be different. This might cause a new device to be created in the backend when using this method to add a sensor to a device.
+ 
+ @param csSensorID		CommonSense sensor ID for the sensor. Cannot be empty.
+ @param	name			Unique name for the device. Cannot be empty.
+ @param	UUID			UUID for the device. Cannot be empty.
+ @param sessionID		The sessionID of the current user. Cannot be empty.
+ @param error			Reference to an NSError object that will contain error information if an error occurs. Cannot be nil.
+ @result				Whether or not the sensor was successfully added to the device.
+ */
+- (BOOL) addSensorWithID: (NSString *) csSensorID toDeviceWithType: (NSString *) deviceType andUUID: (NSString *) UUID andSessionID: (NSString *) sessionID andError: (NSError **) error;
```

Comment:
@jorishjanssen It could be because I am not used to objective-c style yet, but I think the name of this function makes it hard to get a grasp of what it does at first glance. The reader could miss "toDevice".
I prefer addSensorToDeviceWithSensorID: (NSString *) csSensorID WithDeviceType: (NSString *) deviceType andUUID: (NSString *) UUID andSessionID: (NSString *) sessionID andError: (NSError **) error;